framework/validators/FileValidator.php

```diff
@@ -258,7 +259,12 @@ protected function validateValue($file)
      */
     public function getSizeLimit()
     {
+        // Max POST size minus some bytes for form data 
+        $post_limit = $this->sizeToBytes(ini_get('post_max_size')) - 1024;
```

Comment:
That was my initial though, but @samdark correctly pointed out that it's likely that you have some form data along with the file, so it's correct to warn the user.
Anyway I think it's a border line situation, if the file size is SO close to the php limit it will almost certainly need to be raised.